frontends/web/src/routes/device/bitbox02/passphrase.tsx

```diff
@@ -35,345 +36,360 @@ const INFO_STEPS_DISABLE = 0;
 
 const CONTENT_MIN_HEIGHT = '38em';
 
-interface MnemonicPassphraseButtonProps {
+type TProps = {
     deviceID: string;
 }
 
-interface State {
-    infoStep: number;
-    passphraseEnabled?: boolean;
-    status: 'info' | 'progress' | 'success';
-    understood: boolean;
-}
+type TStatus = 'info' | 'progress' | 'success';
+type TPassphraseEnabled = boolean | undefined;
 
-type Props = MnemonicPassphraseButtonProps & TranslateProps;
+export const Passphrase = ({ deviceID }: TProps) => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const [infoStep, setInfoStep] = useState(0);
+  const [status, setStatus] = useState<TStatus>('info');
+  // Can this be a useRef() ?
+  const [passphraseEnabled, setPassphraseEnabled] = useState<TPassphraseEnabled>();
```

Comment:
true, but I think it needs to be `boolean | undefined` because we want to abort in case something goes wrong when loading the passphrase and don't want to assume that it is false by default for example.

I'll remove the type and use `boolean | undefined` directly.

Reply:
> because we want to abort in case something goes wrong when loading the passphrase and don't want to assume that it is false by default for example.

good point, you mean if for example getDeviceInfo(deviceID) returns success: false?

Reply:
I did not really think about concrete scenarios but I guess so yes. I meant in general we don't want to assume true/false for example if the API call did not initialize the variable yet, or it is undefined for any other reason not just an error response.

Reply:
I suggest leave it as is as it is not really about class->functional change.

Reply:
good point, I'll rebase again.

Reply:
actually, if we keep it as is it will also be `boolean | undefined` just implicitly:
https://stackoverflow.com/questions/68420912/ts2322-in-react-usestate-fuction-dispatchsetstateactionmydatatype-undefin

because we don't pass any default boolean. the type of the `setPassphraseEnabled` will be `Dispatch<SetStateAction<boolean | undefined>>` but yeah lets just change that in the next PR.